### DISCUSSION TITLE
Image alt text customized per instance of image

### DISCUSSION THREAD

**Discussion by anphira** [elementor/elementor#28071] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

When inserting an image using Elementor’s image element, the only option for alternative or “alt text” is to pull from the information assigned in the media library. This causes accessibility issues, since alt text is context sensitive. So a given image has different alt text appropriate depending on where it is inserted and what it is used for.

Please take a look at the [alt text decision tree](https://www.w3.org/WAI/tutorials/images/decision-tree/) from W3C. Images have many different uses and purposes and we can not globally apply only one label to a given image every time it is used.

In the built in block editor, we are able to insert an image and customize the alt text for that particular insertion of the image. Which means if the alt text is changed later in the image library, it will not mess up what was put in with the image. And if the image is used for different purpose in the future, that will also not affect previously inserted images.

Being able to write context sensitive alt text is extremely important to web accessibility.

### Describe the solution you'd like.

Have the ability, like the block editor, to enter alt text per image insertion.

### Describe alternatives you've considered.

No acceptable solution has been found within Elementor.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by aebs90** [comment]
@anphira We created a plugin in the WordPress Plugins Directory that does exactly that: https://wordpress.org/plugins/custom-alt-text-for-elementor-images/

Elementor team, it would be cool if you implement this in Elementor, and you are totally free to copy/paste our plugin to add it to Elementor if that's useful. Is there a specific form where we can submit this contribution?

Thanks!

**Comment by Heink** [comment]
This is essential now. Just heard this is a dealbreaker for using Elementor on some new sites.
